src/appleseed/foundation/utility/minizip/zip.c

```diff
@@ -519,15 +519,17 @@ local ZPOS64_T zip64local_SearchCentralDir(const zlib_filefunc64_32_def* pzlib_f
       break;
 
     for (i=(int)uReadSize-3; (i--)>0;)
-      if (((*(buf+i))==0x50) && ((*(buf+i+1))==0x4b) &&
-        ((*(buf+i+2))==0x05) && ((*(buf+i+3))==0x06))
-      {
-        uPosFound = uReadPos+i;
-        break;
-      }
+    {
+        if (((*(buf + i)) == 0x50) && ((*(buf + i + 1)) == 0x4b) &&
+            ((*(buf + i + 2)) == 0x05) && ((*(buf + i + 3)) == 0x06))
+        {
+            uPosFound = uReadPos + i;
+            break;
+        }
 
-      if (uPosFound!=0)
-        break;
+        if (uPosFound != 0)
```

Comment:
I believe you introduced a bug here. I think the initial logic was correct, there was just an indentation problem that was tripping the compiler (one reason why we're so strict on indentation!)